### PR TITLE
[DC-1054] Adding a pull-request-template

### PR DESCRIPTION
**Pull request by evansuslovich**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/DC-1054

## Addresses
Add a pull request template to encourage PR authors to add meaningful pull request descriptions. 

## Summary of changes
- Kept a simple outline of items we want to include in pull request descriptions
- Wanted to stay away from checklists
- We do not yet have a contributing guide for TDR, so we did not include 
- Many other reminders that we could have included are already embedded in our process (for example, we must do a security risk assessment on any jira tickets and checks must pass before you can merge the PR) 

## Testing Strategy
- Met with the DCJ and data exploration team to discuss the changes
- Posted in the #dsp-roundtable-repo-standards channel to allow for further discussion

### CODE REVIEW

**Review by rjohanek — APPROVED** [review]
I like this idea thanks Evan!

**Comment by rushtong** [comment]
I do have some opinions/thoughts but there's no ticket for this, so it's not clear what the requirements are supposed to be. I would suggest iterating over requirements in a Jira ticket before this PR moves forward.
* DSP did have a standard PR template ages ago, but individual repos have diverged significantly over time
* I'm surprised that https://github.com/DataBiosphere/terra-java-project-template doesn't have one of these, it would be better if it did and other repos could derive from that one
* There should be a link to a contributing doc
* There should be a description of merge instructions (tests, thumbs, PO sign-off, etc)
* There should be something about a security risk assessment here - this is in several other repos
  * https://github.com/DataBiosphere/consent/blob/develop/.github/PULL_REQUEST_TEMPLATE.md
  * https://github.com/broadinstitute/firecloud-orchestration/blob/develop/.github/PULL_REQUEST_TEMPLATE.md
  * https://github.com/broadinstitute/sam/blob/develop/.github/PULL_REQUEST_TEMPLATE.md
  * https://github.com/DataBiosphere/leonardo/blob/develop/.github/pull_request_template.md

**Comment by fboulnois** [comment]
> I do have some opinions/thoughts but there's no ticket for this, so it's not clear what the requirements are supposed to be. I would suggest iterating over requirements in a Jira ticket before this PR moves forward.
> [...]

I think that content should live in the original JIRA ticket since that is ultimately the source of truth for the org, and not necessarily in the GitHub PR. In my opinion, here's the minimum I expect to see in a PR:

- A descriptive title
- A link to the ticket
- A description summarizing the changes

For a more complicated ticket, it may be worth adding other things like testing strategies and merge instructions. Here's what the DUOS one looks like:

https://github.com/DataBiosphere/duos-ui/blob/develop/.github/PULL_REQUEST_TEMPLATE.md

In reality, 99% of the time we only use the `Addresses <ticket>` and `Summary <description>` sections.

File: .github/pull_request_template.md
Comment (okotsopoulos) [inline .github/pull_request_template.md]: Storybook is only used for UI development (and specifically terra-ui, not sure what other UI repos have adopted it) so I'd remove this block.

**Review by okotsopoulos — COMMENTED** [review]
Thanks for taking this initiative, @evansuslovich !

I agree that it would be good to have a Jira ticket associated with this work even though it is a documentation-only change (I am trying to be more diligent about this myself).  I'd approve the PR if it did.

I agree with @rushtong that a PR template… template? 😄 seems like something that should be available upstream in a common place.  As a follow-on, it could offer clarity to connect with #dsp-roundtable-repo-standards ([Confluence](https://broadworkbench.atlassian.net/wiki/spaces/TLR/pages/2952298505/GitHub+Repo+Standards)) who have been developing standards for DSP's repositories, which includes documentation standards.  @pshapiro4broad is involved in that effort if I remember correctly.

Right now, TDR does not adhere to those documentation standards:
- Ex. there is no CONTRIBUTING.md

…and in some areas it's not clear whether something is actually a standard at all:
- Ex. the security risk designation is something that I've seen in a small handful of repos, but not most of them, so I wouldn't vote in favor of adding that to this one right now.

I think it would still give us value to have a first pass at a PR template while we explore the rest.

**Comment by rushtong** [comment]
> …and in some areas it's not clear whether something is actually a standard at all:
> 
> * Ex. the security risk designation is something that I've seen in a small handful of repos, but not most of them, so I wouldn't vote in favor of adding that to this one right now.

The security risk assessment comes from the Shared SDLC v2 document, Software Development -> bullet 3 -> sub-bullet c. DUOS had to go through this process which is why we have this in our PR process.

**Review by okotsopoulos — APPROVED** [review]
Thanks for adding the ticket!

Please see my comment about Storybook: https://github.com/DataBiosphere/jade-data-repo/pull/1674/files#r1589260031

I think this is otherwise okay to merge even as we elect to iterate on it further.

**Review by fboulnois — APPROVED** [review]
Current state lgtm 👍

**Review by rjohanek — APPROVED** [review]

File: .github/pull_request_template.md
Comment (okotsopoulos) [inline .github/pull_request_template.md]: Would you mind tweaking this a bit further?

The need for this comment comes from the fact that this is a repo where (presently) both DCJ and DE teams are actively working.  The current CODEOWNERS settings have both teams set as owners, so the flip side is that DE team gets requested to review DCJ PRs too. Our repository norm is that we don't need to wait on the other team's approval to merge unless explicitly requested, in either direction.

**Review by okotsopoulos — APPROVED** [review]
Thanks for taking this to the finish line!

I left a request on the wording for the comment.

File: .github/pull_request_template.md
Comment (snf2ye) [reply .github/pull_request_template.md]: Ah, I didn't realize that we were getting randomly assigned to PRs as well. 
I made a slight tweak to the comment... I'm not sure how much specific detail I want to add to the comment. What do you think of the new version?

Comment (okotsopoulos) [reply .github/pull_request_template.md]: I think that's closer, and in the name of not letting perfect be the enemy of good would elect to merge it as-is :-) I do agree with the instinct not to overexplain in the comment, as that would be liable to get out of date.  We can always adjust the comment down the road if we want.

**Pull request merged**
